rclpy/rclpy/context.py

```diff
@@ -12,20 +12,37 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+from abc import abstractmethod
 from inspect import ismethod
 import sys
 import threading
-from types import TracebackType
+from types import MethodType, TracebackType
 from typing import Callable
 from typing import ContextManager
 from typing import List
 from typing import Optional
 from typing import Type
-import weakref
+from typing import Union
+from weakref import WeakMethod
+
+from rclpy.destroyable import DestroyableType
+
+
+class ContextHandle(DestroyableType):
+    @abstractmethod
+    def ok(self) -> bool:
+        ...
+
+    @abstractmethod
+    def get_domain_id(self) -> int:
+        ...
+
+    def shutdown(self) -> None:
+        ...
```

Comment:
why `shutdown` does not have decorator `@abstractmethod`, i think this needs to be implemented?

```suggestion
    @abstractmethod
    def shutdown(self) -> None:
        ...
```

Reply:
I had misinterpreted  the `Protocol` [documentation](https://typing.readthedocs.io/en/latest/spec/protocol.html#merging-and-extending-protocols). I thought we need `@abstractmethod` for functions that didn't return None however I actually needed to inherit from `Protocol` again. From my understanding `@abstractmethod` would imply `_rclpy.Context()` is some sort of abstract class itself.